src/envoy/http/jwt_auth/pubkey_cache.h

```diff
@@ -80,8 +80,12 @@ class PubkeyCacheItem : public Logger::Loggable<Logger::Id::filter> {
                              // inline jwks never expires.
                              std::chrono::steady_clock::time_point::max());
       if (status != Status::OK) {
-        ENVOY_LOG(warn, "Invalid inline jwks for issuer: {}, jwks: {}",
-                  jwt_config_.issuer(), inline_jwks);
+        ENVOY_LOG(warn,
+                  "Invalid inline jwks for issuer: {}, jwks: {}, error: {}",
+                  jwt_config_.issuer(), inline_jwks, StatusToString(status));
+        throw EnvoyException(fmt::format(
```

Comment:
This is still going to be throw exception in worker thread as it's called in the constructor right?

Reply:
I have the same feeling since the thread local `JwtAuthStore` which owns `PubkeyCache` which construct PubkeyCacheItem on per worker thread. I didn't see the one catching this exception.
https://github.com/istio/proxy/blob/d70d7779ec3b067a43e9439dd9f925395dcdac20/src/envoy/http/jwt_auth/auth_store.h#L71

Reply:
IMHO line 77 could also throw exception

Reply:
I added dummy_store_ in auth_store.h  line 91.   Its purpose is to create a pubkey cache in main thread (config thread).  This store is not used,  it just want to parse the config and validate it.  If anything is wrong, the exception will  be throw in the main thread which will be catch by Envoy LDS subscriber config_update.  The bad LDS will be discarded.

For an invalid LDS config,  it will not be used,  so the exception will NOT be throw in the worker thread.

Reply:
@qiwzhang Oh, I saw it and the comment but I didn't realize it is addressing the thread model issue.   I think it is tricky but it should works.

Reply:
Thanks for the explanation